pkg/controller/resourcecleanup/resourcecleanup_controller.go

```diff
@@ -0,0 +1,296 @@
+package resourcecleanup
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/stolostron/managedcluster-import-controller/pkg/constants"
+	"github.com/stolostron/managedcluster-import-controller/pkg/helpers"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	kevents "k8s.io/client-go/tools/events"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var log = logf.Log.WithName(ControllerName)
+
+// ReconcileResourceCleanup reconciles a ManagedCluster object
```

Comment:
let's add more comments on this reconciler. This will be the main loop to cleanup resources right. We will need a detailed description on the workflow.

Reply:
added